modules/tinymce/src/plugins/table/main/ts/actions/TableActions.ts

```diff
@@ -19,12 +19,14 @@ import * as TableSize from '../queries/TableSize';
 import { ephemera } from '../selection/Ephemera';
 import { getCellsFromSelection, getRowsFromSelection } from '../selection/TableSelection';
 
-type TableAction<T> = (table: SugarElement<HTMLTableElement>, target: T) => Optional<Range>;
+type TableAction<T> = (table: SugarElement<HTMLTableElement>, target: T) => Optional<TableActionResult>;
+export type TableActionResult = { rng: Range; effect: TableEventData };
```

Comment:
These 2 new types really should be interfaces. That said given the rest are types I can sort of understand why it was done like this.

We also should have them as readonly properties.